pandas/io/parsers/python_parser.py

```diff
@@ -875,6 +875,12 @@ def _search_replace_num_columns(
                     or search not in x
                     or (self._no_thousands_columns and i in self._no_thousands_columns)
                     or not self.num.search(x.strip())
+                    or (
+                        self.columns
+                        and self.dtype
+                        and self.columns[i] in self.dtype
+                        and self.dtype[self.columns[i]] is str
```

Comment:
This is to specific, we have to something like ``not is_numeric_dtype(self.dtype.get(self.columns[I])``

Columns should already be defined here

Reply:
Thank you @phofl, I corrected line number 882.

Reply:
Hi @phofl. Could you please take a look at my last commit? Looks like CI failures are unrelated to my changes.

Reply:
I'll have to look more closely, might get to it tomorrow.